client/landing/stepper/hooks/use-change-site-domain-if-needed.ts

```diff
@@ -5,12 +5,12 @@ import { useSiteData } from './use-site-data';
 
 const FREE_DOMAIN_SUFFIX = '.wordpress.com';
 
-const useChangeSiteDomain = () => {
+const useChangeSiteDomainIfNeeded = () => {
 	const dispatch = useDispatch();
 	const { siteId, siteSlug } = useSiteData();
 
-	const changeSiteDomain = async ( domain: string ) => {
-		if ( domain === siteSlug || ! domain.endsWith( FREE_DOMAIN_SUFFIX ) ) {
+	const changeSiteDomainIfNeeded = async ( domain: string ) => {
+		if ( ! siteSlug || domain === siteSlug || ! domain.endsWith( FREE_DOMAIN_SUFFIX ) ) {
```

Comment:
Should we add a check for `siteId` as well? From what I've seen it was 0 while going through the newsletter flow.

Reply:
Before the site is created, both `siteId` (0) and `siteSlug` ('') are falsy. So, I think checking either of them is enough. Does it makes sense?

Reply:
Yes, thank you for claryfing!